packages/schemas/lib/events/eventTypes.ts

```diff
@@ -12,7 +13,8 @@ export function isCommonEventDefinition(entity: unknown): entity is CommonEventD
 
 export type CommonEventDefinition = {
   consumerSchema: ZodObject<
-    Omit<(typeof CONSUMER_BASE_EVENT_SCHEMA)['shape'], 'payload'> & {
+    Omit<(typeof CONSUMER_BASE_EVENT_SCHEMA)['shape'], 'payload' | 'metadata'> & {
```

Comment:
ensuring that metadata here and in baseMessageSchema come from the same place make IDE much happier about infering types for whatever reason